### ISSUE TITLE
`.set_scale()` has no effect for some backends

### ISSUE THREAD

**Issue by jbusecke** [pyvista/pyvista#3695]
### Describe the bug, what's wrong, and what you expected.

I just noticed that the behavior of `plotter.set_scale()` differs depending on the jupyter backend selected.

Only for 'ipyvtklink' I get the desired 'squishing' of a simple sphere if I set one axis scale to 0.5:
<img width="646" alt="image" src="https://user-images.githubusercontent.com/14314623/206206640-58234fa0-f0cb-45d8-b50a-0ee368e92b4c.png">

For all other backends I tested the setting seems to have no effect (here the `panel` backend as example:
<img width="356" alt="image" src="https://user-images.githubusercontent.com/14314623/206206865-7369c90c-41ca-43e6-9381-28df2fd4f9db.png">

### Steps to reproduce the bug.

I was able to reproduce this behavior by commenting/uncommenting the backends in the first block below and rerunning the code in a jupyterlab cell

```python
import pyvista as pv

pv.set_jupyter_backend('ipyvtklink') # works
# pv.set_jupyter_backend('pythreejs') # doesnt
# pv.set_jupyter_backend('panel') #doesnt
# pv.set_jupyter_backend('ipygany') # could not test this

mesh = pv.Sphere()

p = pv.Plotter()
actor = p.add_mesh(mesh)

p.set_scale(zscale=0.5)
p.show()
```

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Wed Dec 07 15:32:05 2022 CET

                OS : Darwin
            CPU(s) : 8
           Machine : arm64
      Architecture : 64bit
               RAM : 16.0 GiB
       Environment : Jupyter
       File system : apfs
        GPU Vendor : Apple
      GPU Renderer : Apple M1
       GPU Version : 4.1 Metal - 76.3

  Python 3.10.8 | packaged by conda-forge | (main, Nov 22 2022, 08:25:29)
  [Clang 14.0.6 ]

           pyvista : 0.37.0
               vtk : 9.2.2
             numpy : 1.23.5
           imageio : 2.22.4
            scooby : 0.7.0
             pooch : v1.6.0
        matplotlib : 3.6.2
           IPython : 8.7.0
        ipyvtklink : 0.2.3
              tqdm : 4.64.1
        jupyterlab : 3.5.1
         pythreejs : Version unknown
--------------------------------------------------------------------------------
```

**Comment by banesullivan** [comment]
tldr; we need to change the internal mechanism we use for scaling in PyVista

AFAIK, there is not equivalent scaling mechanism in VTK.js (see https://github.com/Kitware/vtk-js/issues/2692) to our current camera scaling mechanism in PyVista.

Further, scaling the camera like we do is riddled with issues that affect camera movements, lighting, and the way any text is displayed. (see also #2993)

With this in mind, I strongly advise anyone needing to scale a visualization to scale the points/spatial reference of the data itself -- effectively changing the units of the data. This is a much safer and more intuitive solution IMO -- though it may have a computational cost for large and/or time series datasets.

There is an alternative that may be a more robust solution, which is scaling the actors (this is also available in VTK.js)

```py
import pyvista as pv

mesh = pv.Sphere()

pl = pv.Plotter()
actor = pl.add_mesh(mesh)
actor.scale = (1, 1, 2)  # scale Z axis by 2
pl.show()
```

I wonder if we could internally refactor `set_scale()` to scale all the actors in the renderer automatically. I'll investigate this and see if I can get it working in a robust way.

**Comment by banesullivan** [comment]
See https://github.com/pyvista/pyvista/pull/3990
